pydrive2/files.py

```diff
@@ -292,6 +348,53 @@ def download(fd, request):
                 if boms:
                     self._RemovePrefix(fd, boms[0])
 
+    @LoadAuth
+    def GetContentIOBuffer(
+        self, mimetype=None, encoding=None, chunksize=DEFAULT_CHUNK_SIZE
+    ):
+        """Get a file-like object which has a buffered read() method.
+
+    :param mimetype: mimeType of the file.
+    :type mimetype: str
+    :param encoding: The encoding to use when decoding the byte string.
+    :type encoding: str
+    :param chunksize: default read() chunksize.
+    :type chunksize: int
+    :returns: MediaIoReadable -- file-like object.
+    :raises: ApiRequestError, FileNotUploadedError
+    """
+        files = self.auth.service.files()
+        file_id = self.metadata.get("id") or self.get("id")
+        if not file_id:
+            raise FileNotUploadedError()
+
+        # Should use files.export_media instead of files.get_media if
+        # metadata["mimeType"].startswith("application/vnd.google-apps.").
+        # But that would first require a slow call to FetchMetadata().
+        # We prefer to try-except for speed.
+        try:
+            request = self._WrapRequest(files.get_media(fileId=file_id))
+            return MediaIoReadable(
+                request, encoding=encoding, chunksize=chunksize
+            )
```

Comment:
@shcheklein did you mean you'd prefer:

- `mode="rb" or "r"` as a read-only property of `MediaIoReadable` objects, or
- `mode` as an optional argument to `MediaIoReadable.__init__`?

It seems odd since

- get/download implies `r`
- `a` and `w` don't make sense as alternative modes
- `encoding=None` is the same as `b`

Reply:
Having a text-like mode (py2 `mode="rU"`) would need something like `encoding="utf-8"` and `.decode(encoding).sub("\r\n", "\n")`

Reply:
I'm referring to the need to support this interface: https://dvc.org/doc/api-reference/open

Which takes a lot from the regular `open` - https://docs.python.org/3/library/functions.html#open

E.g. 

```In text mode, if encoding is not specified the encoding used is platform dependent: locale.getpreferredencoding(False) is called to get the current locale encoding. (For reading and writing raw bytes use binary mode and leave encoding unspecified.) 
```

```
The type of file object returned by the open() function depends on the mode. When open() is used to open a file in a text mode ('w', 'r', 'wt', 'rt', etc.), it returns a subclass of io.TextIOBase (specifically io.TextIOWrapper). When used to open a file in a binary mode with buffering, the returned class is a subclass of io.BufferedIOBase. The exact class varies: in read binary mode, it returns an io.BufferedReader; in write binary and append binary modes, it returns an io.BufferedWriter, and in read/write mode, it returns an io.BufferedRandom. When buffering is disabled, the raw stream, a subclass of io.RawIOBase, io.FileIO, is returned.
```

etc

we can either support this logic here, or we'll need to support it on the DVC level.

Reply:
updated description now. Added an `__iter__()` so this can be wrapped à la https://github.com/iterative/dvc/blob/516d82e41107438756797611bbb15614cd4d989f/dvc/utils - not sure if it's doing that wrapping directly in pydrive?